Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -75,6 +80,41 @@ public FreezeOnMatchCustomization(Type targetType)
         public IRequestSpecification Matcher { get; }
 
         /// <summary>
+        /// Used to store a request for resolving a specimen before freezing.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// By default the TargetType property is used as the request, but it can be 
+        /// adjusted to return any other type of request if necessary.
+        /// </para>
+        /// <para>
+        /// A typical use case is to define a request which includes significant 
+        /// information about a specimen. For instance, <see cref="ParameterInfo"/> 
+        /// which is widely used in Glue Libraries, can contain custom attributes
+        /// such as <see cref="StringLengthAttribute"/> or <see cref="RegularExpressionAttribute" />.
+        /// These attributes are essential part of the request since they influence 
+        /// the final specimen. In order not to loose this info, the <see cref="ParameterInfo"/>
+        /// should be set into the Request property.
+        /// </para>
+        /// </remarks>
+        public object Request
+        {
+            get
+            {
+                return this.request;
+            }
+            set
```

Comment:
It seems to me that [resolving the request as `object`](https://github.com/AutoFixture/AutoFixture/pull/695/files#diff-ff3511dac1948f3316dc6bf6343595eaL103) makes `TargetType` obsolete, so there is no need to keep it around. Instead, we could add a constructor that accepts the `request` object and set it that way instead of using a property setter.